state/machine.go

```diff
@@ -2104,6 +2092,61 @@ func (m *Machine) CreateUpgradeSeriesPrepareLock() error {
 	return nil
 }
 
+// RemoveUpgradeSeriesPrepareLock remove a series upgrade prepare lock for a
+// given machine.
+func (m *Machine) RemoveUpgradeSeriesPrepareLock() error {
+	buildTxn := func(attempt int) ([]txn.Op, error) {
+		if attempt > 0 {
+			if err := m.Refresh(); err != nil {
+				return nil, errors.Trace(err)
+			}
+		}
+		locked, err := m.IsLocked()
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		if !locked {
+			return nil, errors.NotFoundf("upgrade series prepare lock for machine %q", m)
```

Comment:
since the doc isn't there... shouldn't we return jujutxn.ErrNoOperations?

Reply:
What is best practice here.